frontend-react/src/components/shared/Table/Table.tsx

```diff
@@ -0,0 +1,225 @@
+import classnames from "classnames";
+import sanitizeHtml from "sanitize-html";
+import { useMemo, useState } from "react";
+import { Icon } from "@trussworks/react-uswds";
+
+import { convertStringToSortableRepresentation } from "../../../utils/misc";
+
+import styles from "./Table.module.scss";
+
+const FilterOptionsEnum = {
+    NONE: "none",
+    ASC: "asc",
+    DESC: "desc",
+} as const;
+
+interface SortableTableHeaderProps {
+    columnTitle: string;
+    activeColumn: string;
+    sortOrder: string;
+    setSortOrder: (order: string) => void;
+    setActiveColumn: (column: string) => void;
+}
+
+interface RowData {
+    columnKey: string;
+    content: string;
+    modalContent?: string;
```

Comment:
We chatted about this before, but I don't think `modalContent` is necessarily something we want to build into the table.  I think it might be better to break this down into different pieces:
- `displayValue`: the value that's displayed on the cell prior to any interaction
- `rawValue`: the value we get back verbatim (could be equivalent to `displayValue` in many cases)
- `CellComponent`: custom component that would render the values
    - By default, we can make a component that renders `rawValue` in a modal

The idea here is to ensure that we're as flexible as we need to be with the different values so the table can easily scale to different display capabilities.

Reply:
So if we want to utilize the modal in the future, each cell would have to have a instance of a modal?